### DISCUSSION TITLE
Import modules via npm import

### DISCUSSION THREAD

**Discussion by nikolailehbrink** [tailpress/tailpress#125] [Q&A] [answered]
Hello, unfortunately I can not get any modules working via "import" and npm install. Here is an example for [swiper.js](https://swiperjs.com/get-started)

`npm install swiper`

In app.js:
```
// import swiper bundle with all installed modules
import swiper from "swiper/bundle";

// import styles bundle
import "swiper/css/bundle";

// initialize
const swiper = new Swiper(".swiper", {
  // Optional parameters
  direction: "vertical",
  loop: true,

  // If we need pagination
  pagination: {
    el: ".swiper-pagination",
  },

  // navigation arrows
  navigation: {
    nextEl: ".swiper-button-next",
    prevEl: ".swiper-button-prev",
  },

  // And if we need a scrollbar
  scrollbar: {
    el: ".swiper-scrollbar",
  },
});
```
Then in single.php:

```
<!-- Slider main container -->
<div class="swiper">
  <!-- Additional required wrapper -->
  <div class="swiper-wrapper">
    <!-- Slides -->
    <div class="swiper-slide">Slide 1</div>
    <div class="swiper-slide">Slide 2</div>
    <div class="swiper-slide">Slide 3</div>
    ...
  </div>
  <!-- If we need pagination -->
  <div class="swiper-pagination"></div>

  <!-- If we need navigation buttons -->
  <div class="swiper-button-prev"></div>
  <div class="swiper-button-next"></div>

  <!-- If we need scrollbar -->
  <div class="swiper-scrollbar"></div>
</div>
```

I also tried the almost same procedure for the [aos-library](https://michalsnik.github.io/aos/) but that didn't work as well. Would be nice, if you could give me an example on how to do it :)

**Comment by jeffreyvr** [comment, marked as answer]
This is setup from the Swiper documentation works for me:

`resources/js/app.js`

```js
import Swiper, { Navigation, Pagination } from 'swiper';
import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';
```

```js
window.addEventListener('load', function () {
      // ...
      const swiper = new Swiper('.swiper', {
            // configure Swiper to use modules
            modules: [Navigation, Pagination]
      });
});
```

**Reply by nikolailehbrink** [reply]
Hi Jeffrey, thank you for your answer, could you try that out with aos once for me please. I don't know how to configure it via @import. 
Would be very nice.

**Reply by charlie-george-1989** [reply]
@nikolailehbrink Did you find a fix for this? I'm experiencing the same issues with VenoBox and Magnific.

**Reply by charlie-george-1989** [reply]
Swiper also works for me, as well as Three.js. But not the above 2 mentioned for some reason. Importing like this:

import VenoBox from 'venobox';

But VSCode is underlining 'venobox' with the 3 dots warning:

"Could not find a declaration file for module 'venobox'. '/Users/charlie/Charlie/MyDev/Clients/fred-james/local/app/public/wp-content/themes/fred/node_modules/venobox/dist/venobox.min.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/venobox` if it exists or add a new declaration (.d.ts) file containing `declare module 'venobox';`ts(7016)."

**Reply by nikolailehbrink** [reply]
@charlie-george-1989 Hi Charlie, the last warning is because you are using typescript and the types for these packages are missing. Have you tried out to run the proposed command `npm i --save-dev @types/venobox`?

**Reply by nikolailehbrink** [reply]
I just looked into the package you were refering to and it appears to have no types as it is written in JS not TS. Here is how you can get rid of the warning: https://stackoverflow.com/questions/46608937/npm-package-doesnt-have-types

**Reply by charlie-george-1989** [reply]
@nikolailehbrink Hi mate, thanks for the tip, I think I have already tried but will give it another go and report back.

**Reply by charlie-george-1989** [reply]
Oh okay! Brill, I'll take a look, thank you.

**Reply by charlie-george-1989** [reply]
None of those methods are working for me. I made a new globals.d.ts file in the root director and have added:

declare module 'venobox';
declare module 'magnific-popup';

But that isn't working. I restarted my local server and recompiled, to no avail. Seems crazy how TailPress isn't liking npm installs.
